### ISSUE TITLE
Reposlite throw error / reset repository when version end with dot(.)

### ISSUE THREAD

**Issue by milkyway0308** [dzikoysk/reposilite#478]
Hello again, today I got mistake, and accidentally not typed version on end, and intellij deployed with 1.1. version.
It made Reposilite throws error, cause I don't know why, cannot fix it.
My server is running on Ubuntu 20.04.2, with Java 11.

Here's stack trace.

```
00:00:05.340 ERROR | : java.nio.file.DirectoryNotEmptyException: /home/skywolf46/reposilite/./repositories.lock
        at java.base/sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:247)
        at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
        at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
        at java.base/java.nio.file.Files.copy(Files.java:3055)
        at org.panda_lang.reposilite.repository.RepositoryStorage.storeFile(RepositoryStorage.java:117)
        at org.panda_lang.reposilite.repository.RepositoryStorage.storeFile(RepositoryStorage.java:94)
        at org.panda_lang.reposilite.repository.RepositoryService.lambda$storeFile$2(RepositoryService.java:70)
        at org.panda_lang.reposilite.repository.RepositoryService.tryExecute(RepositoryService.java:82)
        at org.panda_lang.reposilite.repository.RepositoryService.storeFile(RepositoryService.java:69)
        at org.panda_lang.reposilite.repository.DeployService.deploy(DeployService.java:90)
        at org.panda_lang.reposilite.repository.DeployEndpoint.handle(DeployEndpoint.java:76)
        at io.javalin.core.security.SecurityUtil.noopAccessManager(SecurityUtil.kt:23)
        at io.javalin.http.JavalinServlet$addHandler$protectedHandler$1.handle(JavalinServlet.kt:121)
        at io.javalin.http.JavalinServlet$service$2$1.invoke(JavalinServlet.kt:45)
        at io.javalin.http.JavalinServlet$service$2$1.invoke(JavalinServlet.kt:24)
        at io.javalin.http.JavalinServlet$service$1.invoke(JavalinServlet.kt:129)
        at io.javalin.http.JavalinServlet$service$2.invoke(JavalinServlet.kt:40)
        at io.javalin.http.JavalinServlet.service(JavalinServlet.kt:81)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at io.javalin.websocket.JavalinWsServlet.service(JavalinWsServlet.kt:51)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:791)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:550)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
        at io.javalin.core.JavalinServer$start$wsAndHttpHandler$1.doHandle(JavalinServer.kt:49)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
        at org.eclipse.jetty.server.Server.handle(Server.java:516)
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:279)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by milkyway0308** [comment]
P.S. After this error, my repository <b>deleted</b>.
Maybe it's major bug, need to fix.

Screenshot - 
https://imgur.com/a/WAaL3tv

**Comment by milkyway0308** [comment]
PPS. My repository live on repository.lock directory! I'm happy that I don't need re-uploading of artifacts.

**Comment by dzikoysk** [comment]
Thank you for reporting this. It's a serious problem, but fortunately it may only happen for invalid data provided by authenticated user. I'm really busy right now, but I'll probably provide a fix tonight.

**Comment by milkyway0308** [comment]
Always thanks for maintenance Reposilite!
I always satisfied on Reposilite, it greatly increased my development speed.

Not a lot, but I decided to support periodically to panda-lang.
Hope it help!

**Comment by dzikoysk** [comment]
Thanks :) 

Fix is available in [Reposilite 2.9.22](https://github.com/dzikoysk/reposilite/releases/tag/2.9.22), I'm glad you didn't lost the content of your repository.
